fury/utils.py

```diff
@@ -882,100 +882,99 @@ def normals_from_v_f(vertices, faces):
+    m_orient = np.ones((4, 4))
+    m_orient[0, :3] = v1
+    m_orient[1, :3] = v2
+    m_orient[2, :3] = v3
+    m_orient[3, :3] = 0
 
-    val = (y2 - y1) * (x3 - x2) - (y3 - y2) * (x2 - x1)
+    val = np.linalg.det(m_orient)
 
-    if val < 0:
-        return 1
-    return 0
+    return bool(val > 0)
 
 
-def change_order(tri):
-    """ Changes the order of a given triangle
+def change_vertices_order(triangle):
+    """Change the vertices order of a given triangle.
 
-    Parameter
-    ---------
-    tri: ndarray
+    Parameters
+    ----------
+    triangle : np.ndarray, shape(1, 3)
         array of 3 vertices making up a triangle
 
     Returns
     -------
-    np.array(newVert): ndarray
-        new array of vertices making up a triangle in the opposite winding\
+    new_triangle: np.ndarray, shape(1, 3)
+        new array of vertices making up a triangle in the opposite winding
         order of the given parameter
-    """
 
-    newVert = [tri[2], tri[1], tri[0]]
+    """
+    return np.array([triangle[2], triangle[1], triangle[0]])
 
-    return np.array(newVert)
 
+def fix_winding_order(vertices, triangles, clockwise=False):
+    """Return corrected triangles.
 
-def check_order(vert, triarr):
-    """
+    Given an ordering of the triangle's three vertices, a triangle can appear
+    to have a clockwise winding or counter-clockwise winding.
+    Clockwise means that the three vertices, in order, rotate clockwise around
+    the triangle's center.
 
-    Parameter
-    ---------
-    vert: ndarray
+    Parameters
+    ----------
+    vertices: np.ndarray
         array of vertices corresponding to a shape
-    triarr: ndarray
+    triangles: np.ndarray
         array of triangles corresponding to a shape
+    clockwise : bool
+        triangle order type: clockwise (default) or counter-clockwise.
 
     Returns
     -------
-    correct_vert: ndarray
+    corrected_triangles: np.ndarray
         The corrected order of the vert parameter
 
     """
-
-    shape = triarr.shape
-    correct_vert = np.empty(shape)
-    correct_order = what_order(vert, triarr[0])
-    for nb, i in enumerate(triarr):
-        order2 = what_order(vert, i)
-        if correct_order != order2:
-            temp = change_order(i)
-            correct_vert[nb] = temp
-        else:
-            correct_vert[nb] = i
-    return correct_vert
+    corrected_triangles = triangles.copy()
+    desired_order = clockwise
+    for nb, face in enumerate(triangles):
+        current_order = triangle_order(vertices, face)
+        if desired_order != current_order:
+            corrected_triangles[nb] = change_vertices_order(face)
+    return corrected_triangles
 
 
 def vertices_from_actor(actor):
     """Return vertices from actor.
 
     Parameters
-    -------------
+    ----------
     actor : actor
 
     Returns
-    ---------
-    vertices : ndarray
+    -------
+    vertices : np.ndarray
```

Comment:
We always say ndarray why to say now np.ndarray?

Reply:
using the `np.`before permit to be recognize by intersphinx and the user can click and be forwarded to the right documentation (here, the numpy doc)

Reply:
Clearly not necessary - everyone knows numpy these days

Reply:
I can not agree with this statement. I would say everybody knows how to go to NumPy doc. 
Instead of doing a google search, this notation forwards you automatically to the right place in Numpy doc. 
I will put it back even if I do not agree with your previous statement.

Reply:
Well if you feel very strong about it you can make another PR where you change this everywhere not only here.